tuf/data/types_test.go

```diff
@@ -176,3 +177,46 @@ func TestCheckValidHashStructures(t *testing.T) {
 	err = CheckValidHashStructures(hashes)
 	require.IsType(t, ErrInvalidChecksum{}, err)
 }
+
+func TestCompareMultiHashes(t *testing.T) {
+	var err error
+	hashes1 := make(Hashes)
+	hashes2 := make(Hashes)
+
+	// Expected to fail since there are no checksums at all
+	err = CompareMultiHashes(hashes1, hashes2)
+	require.Error(t, err)
+
+	// Expected to fail even though the checksum of sha384 is valid,
+	// because we haven't provided a supported hash algorithm yet (ex: sha256).
+	hashes1["sha384"], err = hex.DecodeString("64becc3c23843942b1040ffd4743d1368d988ddf046d17d448a6e199c02c3044b425a680112b399d4dbe9b35b7ccc989")
```

Comment:
I think I'm fine with leaving the behavior as is to match `CheckHashes`.

My original thought was that since this is checking equivalency, it'd make sense that if both targets specified some other unsupported hash, that they should be the same, but you're right that it doesn't necessarily make sense since someone might be using the hash structure to store data for so me other random feature that we don't support.